edxval/migrations/0010_remove_video_id.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
```

Comment:
Reminder: remove this older migration file and regenerate a new one of version 10.

Reply:
@nasthagiri I did it deliberately, as with combined migration I was getting error on running the migration on database.

Combined migration:
```
# -*- coding: utf-8 -*-
from __future__ import unicode_literals

from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('edxval', '0009_auto_20171127_0406'),
    ]

    operations = [
        migrations.AddField(
            model_name='videotranscript',
            name='video',
            field=models.ForeignKey(related_name='video_transcripts', default=b'', to='edxval.Video'),
        ),
        migrations.AlterUniqueTogether(
            name='videotranscript',
            unique_together=set([('video', 'language_code')]),
        ),
        migrations.RemoveField(
            model_name='videotranscript',
            name='video_id',
        ),
    ]
```
Got following rrror on running `manage.py migrate`:
```
Rendering model states... DONE
  Applying edxval.0010_auto_20180123_0907...Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/commands/migrate.py", line 222, in handle
    executor.migrate(targets, plan, fake=fake, fake_initial=fake_initial)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/migrations/executor.py", line 110, in migrate
    self.apply_migration(states[migration], migration, fake=fake, fake_initial=fake_initial)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/migrations/executor.py", line 148, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/migrations/migration.py", line 115, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/migrations/operations/fields.py", line 62, in database_forwards
    field,
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/sqlite3/schema.py", line 179, in add_field
    self._remake_table(model, create_fields=[field])
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/sqlite3/schema.py", line 140, in _remake_table
    self.create_model(temp_model)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/base/schema.py", line 286, in create_model
    self.execute(sql, params or None)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/base/schema.py", line 111, in execute
    cursor.execute(sql, params)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/utils.py", line 98, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/utils.py", line 62, in execute
    return self.cursor.execute(sql)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/sqlite3/base.py", line 316, in execute
    return Database.Cursor.execute(self, query)
django.db.utils.OperationalError: duplicate column name: video_id
```
With swapping the `RemoveField` and `AddField` in a migration, I got following error:
```
Rendering model states... DONE
  Applying edxval.0010_auto_20180123_0907...Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/core/management/commands/migrate.py", line 222, in handle
    executor.migrate(targets, plan, fake=fake, fake_initial=fake_initial)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/migrations/executor.py", line 110, in migrate
    self.apply_migration(states[migration], migration, fake=fake, fake_initial=fake_initial)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/migrations/executor.py", line 148, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/migrations/migration.py", line 115, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/migrations/operations/fields.py", line 121, in database_forwards
    schema_editor.remove_field(from_model, from_model._meta.get_field(self.name))
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/sqlite3/schema.py", line 197, in remove_field
    self._remake_table(model, delete_fields=[field])
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/sqlite3/schema.py", line 140, in _remake_table
    self.create_model(temp_model)
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/backends/base/schema.py", line 272, in create_model
    columns = [model._meta.get_field(field).column for field in fields]
  File "/Users/muhammadrehan/.virtualenvs/edx-val/lib/python2.7/site-packages/django/db/models/options.py", line 554, in get_field
    raise FieldDoesNotExist('%s has no field named %r' % (self.object_name, field_name))
django.core.exceptions.FieldDoesNotExist: VideoTranscript has no field named u'video_id'
```
And I think this one is due to altering unique together. So, I thought to have different migrations for 1) alter unique together and remove video_id  2) adding a video foriegn key and alter unique together again.

Reply:
OR we can have single migration like below:
```
operations = [
        migrations.AlterUniqueTogether(
            name='videotranscript',
            unique_together=set([('language_code',)]),
        ),
        migrations.RemoveField(
            model_name='videotranscript',
            name='video_id',
        ),
        migrations.AddField(
            model_name='videotranscript',
            name='video',
            field=models.ForeignKey(related_name='video_transcripts', default=b'', to='edxval.Video'),
        ),
        migrations.AlterUniqueTogether(
            name='videotranscript',
            unique_together=set([('video', 'language_code')]),
        ),
    ]
```